src/test/scala/chisel3/tests/AsyncResetRegTest.scala

```diff
@@ -0,0 +1,159 @@
+// See LICENSE for license details.
+
+package chisel3.tests
+
+
+import chisel3._
+import chisel3.core.IntParam
+import chisel3.tester._
+import chisel3.util.{Cat, HasBlackBoxResource}
+import firrtl.ir.{Param, Type}
+import org.scalatest._
+import treadle.executable.{PositiveEdge, Transition}
+import treadle.{ScalaBlackBox, ScalaBlackBoxFactory}
+
+object AsyncResetReg {
+  val bitWidth: Int = 3
+}
+
+class AsyncResetReg(resetValue: Int = 0)
+        extends BlackBox(Map("RESET_VALUE" -> IntParam(resetValue))) with HasBlackBoxResource {
```

Comment:
We should probably be using ExtModule instead, which has more consistent semantics.

Reply:
Uses ExtModule now